sematic/plugins/kuberay_wrapper/basic.py

```diff
@@ -0,0 +1,340 @@
+# Standard Library
+import json
+from copy import deepcopy
+from typing import Any, Dict, Type
+
+# Sematic
+from sematic.abstract_plugin import AbstractPluginSettingsVar
+from sematic.config.settings import get_plugin_setting
+from sematic.plugins.abstract_kuberay_wrapper import (
+    AbstractKuberayWrapper,
+    RayClusterConfig,
+    RayClusterManifest,
+    RayNodeConfig,
+    ScalingGroup,
+)
+from sematic.utils.exceptions import UnsupportedError, UnsupportedVersionError
+
+
+class BasicKuberaySettingsVar(AbstractPluginSettingsVar):
+    """Settings for the Kubray wrapper.
+
+    Attributes
+    ----------
+    RAY_GPU_NODE_SELECTOR:
+        The Kubernetes node selector that will be used for Ray nodes
+        that use GPUs. Value should be json encoded into a string.
+    RAY_NON_GPU_NODE_SELECTOR:
+        The Kubernetes node selector that will be used for Ray nodes
+        that don't use GPUs. Value should be json encoded into a string.
+    RAY_GPU_TOLERATIONS:
+        The Kubernetes tokerations that will be used for Ray nodes
+        that use GPUs. Value should be json encoded into a string.
+    RAY_NON_GPU_TOLERATIONS:
+        The Kubernetes tokerations that will be used for Ray nodes
+        that don't use GPUs. Value should be json encoded into a string.
+    RAY_GPU_RESOURCE_REQUEST_KEY:
+        The key that will be used in the Kubernetes resource requests/
+        limits fields to indicate how many GPUs are required for Ray
+        head/worker nodes. If not specified, requests for more than one
+        GPU will be denied. Value should be json encoded into a string.
+    RAY_SUPPORTS_GPUS:
+        Whether Ray heads/workers can be assigned to Kubernetes nodes with
+        GPUs. Should be either set to "true" or "false".
```

Comment:
Are these server settings? Can they easily be passed as env vars? Or do we expect users to have settings files on their servers?

Reply:
They are plugin settings. All plugin settings can be overridden by env vars. If you look at the tests, you will see that the way they are configured there is via mocking env vars